### DISCUSSION TITLE
1주차 사전 과제, 장준환

### DISCUSSION THREAD

**Discussion by jun-hwan00** [MrMirror21/2024-KHU-CSE-TrackStudy-Web#20] [General]
- 명명 규칙(Naming Convention)이라는 것이 있습니다. 어떤 규칙들이 있는지, 각 규칙들은 어떻게 코드를 작성하는지, 자신이 그동안 사용해왔던 명명 규칙은 무엇인지 적어주세요.

Camel Case
낙타 표기법. 각 단어의 첫 문자를 대문자로 표기하고 이름의 첫 문자는 소문자로 작성

Pascal Case 
각 단어의 첫 문자를 대문자로 표기하며, 이름이 첫 문자 또한 대문자로 작성

Snake Case 
각 단어의 사이를 언더바로 구분해주는 표기법

저는 보통 Camel Case방법을 통해 변수나 함수등의 이름을 지어줬습니다.

- 함수가 순수하다는 것은 무엇을 의미할까요?

함수가 순수하다는 것은 동일한 인자를 받았을 때 항상 같은 값을 반환하고 부수효과가 없는 것을 의미

- 개발 시에 순수 함수(pure function)로만 모든 코드를 작성하는 것에 대한 생각을 적어주세요.

개발자는 효율적으로 코드를 짜고 빠르게 동작하는 소프트웨어를 만들어야한다. 순수함수만으로 코드를 짠다면 때로는 성능면에서 효율적이지 않을 수 있다. 그렇기때문에 적절하게 섞어서 사용해야 한다고 생각한다.  

자바스크립트에서 함수를 작성하는 문법은 다음과 같습니다. 이 외의 함수를 작성하는 문법 두 가지가 무엇이며, 어떤 때 사용하나요?
```
function myFunc(param) {
	statements
}
```
1. 화살표 표기법
```
const myFunc = (param) => {
};
```
2. 함수 표현식
```
const myFunc = function(param) {
};
```
파이썬에서는 문자열을 포맷팅하기 위해 다음과 같은 f-string을 사용합니다. 자바스크립트에서 f-string과 같이 문자열을 다양한 자료형과 변수들을 혼합하여 사용하려면 어떤 방법을 사용하나요?

``` 
let x = 1;
let y = 2;
console.log(`${x} + ${y}는 ${x+y}입니다.`);
```

다음과 같은 객체(object)에서 p 값을 출력하려면 o.p 와 같은 방식으로 출력해야 하는데, p 로 출력할 수 있는 방법은 없을까요?
```
var o = {p:42, q: true};
var{p} = o;
console.log(p); //42
```
객체 디스트럭처링 방법으로 가능하다

**Comment by MrMirror21** [comment]
답변해주신 사전 과제 잘 봤습니다.
전체적으로 잘 답변해주셨고, 함수 작성법 관해서 각 작성법이 어떤 때에 사용되는지 생각해보는 것도 도움이 될 것 같습니다.
